Sources/Credential/Credential Base Classes/BaseOAuthSwiftCredential.swift

```diff
@@ -130,125 +72,103 @@ open class OAuthSwiftCredential: NSObject, NSCoding {
         coder.encode(self.oauthTokenSecret, forKey: CodingKeys.oauthTokenSecret)
         coder.encode(self.oauthVerifier, forKey: CodingKeys.oauthVerifier)
         coder.encode(self.oauthTokenExpiresAt, forKey: CodingKeys.oauthTokenExpiresAt)
-        coder.encode(self.version.toInt32, forKey: CodingKeys.version)
+        coder.encode(self.version, forKey: CodingKeys.version)
     }
-    // } // End NSCoding extension
-
+    // End NSCoding
+    
     // MARK: functions
     // for OAuth1 parameters must contains sorted query parameters and url must not contains query parameters
     open func makeHeaders(_ url: URL, method: OAuthSwiftHTTPRequest.Method, parameters: OAuthSwift.Parameters, body: Data? = nil) -> [String: String] {
-        if let factory = headersFactory {
-            return factory.make(url, method: method, parameters: parameters, body: body)
-        }
-        switch self.version {
-        case .oauth1:
-            return ["Authorization": self.authorizationHeader(method: method, url: url, parameters: parameters, body: body)]
-        case .oauth2:
-            return self.oauthToken.isEmpty ? [:] : ["Authorization": "Bearer \(self.oauthToken)"]
-        }
+        preconditionFailure("This abstract method must be overridden")
     }
-
+    
     open func authorizationHeader(method: OAuthSwiftHTTPRequest.Method, url: URL, parameters: OAuthSwift.Parameters, body: Data? = nil) -> String {
         let timestamp = String(Int64(Date().timeIntervalSince1970))
-        let nonce = OAuthSwiftCredential.generateNonce()
+        let nonce = BaseOAuthSwiftCredential.generateNonce()
```

Comment:
SignTests.swift compile?
generateNonce is public some same here there is a breaking change, but I do not see any solution